app/uk/gov/hmrc/helptosave/controllers/AccountController.scala

```diff
@@ -29,11 +33,23 @@ class AccountController @Inject() (proxyConnector: HelpToSaveProxyConnector,
                                    authConnector:  AuthConnector)(implicit transformer: LogMessageTransformer, appConfig: AppConfig)
   extends HelpToSaveAuth(authConnector) with WithMdcExecutionContext {
 
-  def getAccount(): Action[AnyContent] = authorised { implicit request ⇒ implicit nino ⇒
-    proxyConnector.getAccount(nino, request.rawQueryString)
-      .fold(
-        e ⇒ InternalServerError(""),
-        account ⇒ Ok(Json.toJson(account))
-      )
+  def getAccount(nino: String, systemId: String, correlationId: Option[String]): Action[AnyContent] = authorised { implicit request ⇒ implicit authNino ⇒
+    if (!uk.gov.hmrc.domain.Nino.isValid(nino)) {
+      logger.warn("NINO in request was not valid")
+      BadRequest
+    } else if (nino =!= authNino) {
+      logger.warn("NINO in request did not match NINO found in auth")
+      Forbidden
+    } else {
+      val id = correlationId.getOrElse(UUID.randomUUID().toString)
+      proxyConnector.getAccount(nino, systemId, id)
+        .fold(
+          { e ⇒
+            logger.warn(e)
+            InternalServerError
+          },
+          _.fold[Result](NotFound)(account ⇒ Ok(Json.toJson(account)))
```

Comment:
if there are any validation errors (but the account still exists) then it returns `NotFound ` to the caller , does this make sense  ?

Reply:
validation errors get mapped to `InternalServerError` - the `None` case here is when NS&I have said there is no account

Reply:
ok cool, 
but how does the caller distinguish between a real ISE and validation errors,
if there are any validation errors, why don't we return something meaningful to the caller and then caller , based on the repose, says "your account is in inconsistent state, please call us...blah...", 
because if user sees ISE, then he might be keep trying to close his account everyday and he says same ISE :) :)

Reply:
that's a valid point - this might be a question for Jean to see if we want to show something different to the user for our use-case when the information coming back from NS&I looks dodgy. The premise for returning a 500 here is that this endpoint in our BE is in charge of looking to see if the data coming back from NS&I looks OK and if so modify the format of the response to something more easy to use. If it turns out we do want to show different content if the data looks dodgy then we can go back and add a JSON body or something to the response so that our FE can pick that up.